### PR TITLE
added function to output vpts in csv format

### PR DESCRIPTION
**Pull request by iskandari**
- `--csv` must should come as the last argument. If added, a vpts file will be generated in CSV format according to [vpts data exchange format](https://aloftdata.eu/vpts-csv/)
-  output "filename.csv" must come as the second to last argument
- no checks implemented to verify that data types match

### CODE REVIEW

File: src/vol2bird.c [hunk 1]
```diff
@@ -341,111 +386,124 @@ int main(int argc, char** argv) {
...
+        profileBio = NULL;
+        free((void *)profileAll);
+        free((void *)profileBio);
 
         //}
     } // getter scope end
 
+    if (formatCSV){
+    
+        writeCSV(fileVpOut, &alldata, source, fileIn[0], date, time, volume);
```
Comment (adokter) [inline src/vol2bird.c:472]: this should be brought down, since fileVpOut might be NULL

File: src/vol2bird.c [hunk 2]
```diff
@@ -341,111 +386,124 @@ int main(int argc, char** argv) {
...
+        // RaveList_t* attnames = VerticalProfile_getAttributeNames(alldata.vp);
+        // RaveObjectList_t* attvalues = VerticalProfile_getAttributeValues(alldata.vp);
+        // printf("Number of names: %d\n", attnames);
+        // printf("Number of values: %d\n", attvalues);
+
+        // save rave profile to ODIM hdf5 file
+        if (fileVpOut != NULL)
+        {
+            int result;
+            result = saveToODIM((RaveCoreObject *)alldata.vp, fileVpOut);
```
Comment (adokter) [inline src/vol2bird.c:491]: suggest inserting writeCSV function here, determining which format to write to based on extension of fileVpOut

File: lib/libvol2bird.c [hunk 3]
```diff
@@ -40,9 +39,10 @@
 #undef RAD2DEG // to suppress redefine warning, also defined in dealias.h
 #undef DEG2RAD // to suppress redefine warning, also defined in dealias.h
 #include "libdealias.h"
-
+#include <assert.h>
```
Comment (adokter) [inline lib/libvol2bird.c:42]: do we need this without the vpts csv format assertions?

File: lib/libvol2bird.h [hunk 4]
```diff
@@ -23,6 +23,8 @@
 
 #define DEG2RAD 0.01745329251994329576 // Degrees to radians.
 #define RAD2DEG (57.29578)    // Radians to degrees.
+#define MAX_DOUBLE DBL_MAX // Allow Inf in max values
```
Comment (adokter) [inline lib/libvol2bird.h:26]: where is this used? does it change behavior of general max function? - Just want to make sure it doesn't change behavior elsewhere

**Review by adokter — CHANGES_REQUESTED** [review]

**Comment by adokter** [comment]
In addition to review comments added above:
1) generated csv contains some unnecessary whitespace from the previous fixed delimited format
2) source_file field should be in quotes, since filenames could theoretically contain a `,` in the filename

**Comment by iskandari** [comment]
Requested changes made, please review:

- Formatting is based on whether `fileVpOut` contains a .csv string
- All whitespace is removed before writing each new line of the CSV (newlines are preserved)
- Validation code removed

**Comment by iskandari** [comment]
Copied latest working versions of libvol2bird.c and libvol2bird.h from main in [vol2birdR repo](https://github.com/adokter/vol2birdR)

**Pull request merged**